file/file_util.cc

```diff
@@ -151,14 +153,22 @@ IOStatus GenerateOneFileChecksum(
         requested_checksum_func_name +
         " from checksum factory: " + checksum_factory->Name();
     return IOStatus::InvalidArgument(msg);
+  } else {
+    // For backward compatibility and use in file ingestion clients where there
```

Comment:
I also updated this comment a bit since it looks like empty string is passed during file ingestion , which appears unrelated to compatibility.